pkg/store/blobstore/casync_store.go

```diff
@@ -46,16 +47,20 @@ func NewCasyncStore(localStoreDir, localIndexStoreDir string) (*CasyncStore, err
 		return nil, err
 	}
 
+	concurrency := runtime.NumCPU()
+	if concurrency > 4 {
+		concurrency = 4
```

Comment:
why is it limited to 4?

Reply:
This is only used for chunking new .nar file uploads - see https://pkg.go.dev/github.com/folbricht/desync#readme-parallel-chunking.

It's not used for serving .nar files.

Right now, nix-casync stores the (uncompressed) uploaded .nar file in a temporary directory, then runs the chunker. It returns on the HTTP request once everything has been uploaded into the chunk store - so this improves the waiting time between the client finished sending the data, and the server finished chunking and uploading.

I'd prefer having a (single-threaded) chunker that can chunk (and upload) the file in a streaming fashion (while it's being uploaded).

I'm not sure how long we'll end up with this code anyways.

I doubt the performance benefits justify the increase of CPU usage, especially considering there might be multiple .nar file uploads going on in parallel.